docs/source/api/commands.rst

```diff
@@ -1,47 +1,641 @@
++------------+-------------------------------------+--------------------------------------------------------+
+| Precision  | number of digits after comma/dot    | 0 <= precision <= uint32 max                           |
++------------+-------------------------------------+--------------------------------------------------------+
+
+Validation
+^^^^^^^^^^
+
+1. Transaction creator has permission to create assets
+2. Asset name is unique per domain
+
 Create domain
 -------------
 
+Purpose
+^^^^^^^
+
+The purpose of create domain command is to make new domain in Iroha network, which is a group of accounts.
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message CreateDomain {
+        string domain_id = 1;
+        string default_role = 2;
+    }
+
+Structure
+^^^^^^^^^
+
++--------------+-----------------------------------------+----------------------------+
+| Field        | Description                             | Constraint                 |
++--------------+-----------------------------------------+----------------------------+
+| Domain ID    | ID for created domain                   | unique, `[0-9A-Za-z]{1,9}` |
++--------------+-----------------------------------------+----------------------------+
+| Default role | role for any created user in the domain | one of the existing roles  |
++--------------+-----------------------------------------+----------------------------+
+
+Validation
+^^^^^^^^^^
+
+1. Domain ID is unique
+2. Account, who sends this command in transaction, has role with permission to create domain
+3. Role, which will be assigned to created user by default, exists in the system
+
 Create role
 -----------
 
+Purpose
+^^^^^^^
+
+The purpose of create role command is to create a new role in the system from the set of permissions.
+Combining different permissions into roles, maintainers of Iroha peer network can create customized security model.
+
+Schema
+^^^^^^
+
+.. code-block:: proto
+
+    message CreateRole {
+       string role_name = 1;
+       repeated string permissions = 2;
+    }
+
+Structure
+^^^^^^^^^
+
++-------------+---------------------------------------+------------------------------------------------------------------------------+
+| Field       | Description                           | Constraint                                                                   |
++=============+=======================================+==============================================================================+
+| Role name   | name of role to create                | `[A-Za-z0-9_]{1,7}`                                                          |
++-------------+---------------------------------------+------------------------------------------------------------------------------+
+| Permissions | array of already existent permissions | set of passed permissions is fully included into set of existing permissions |
```

Comment:
Also, we have to make a policy about dots.

Reply:
about what?

Reply:
?